libs/domain/site/currency-selector/src/currency-selector.component.spec.ts

```diff
@@ -0,0 +1,193 @@
+      element = await fixture(
+        html`<oryx-site-currency-selector></oryx-site-currency-selector>`
+      );
+    });
+
+    it('should not render the currency selector', () => {
+      expect(element).not.toContainElement('button');
+    });
+  });
+
+  describe('when there is only one currency', () => {
+    beforeEach(async () => {
+      currencyService.getAll.mockReturnValue(of([{ code: 'EUR' } as Currency]));
+      element = await fixture(
+        html`<oryx-site-currency-selector></oryx-site-currency-selector>`
+      );
+    });
+
+    it('should not show the currency selector', () => {
+      expect(element).not.toContainElement('button');
+    });
+  });
+
+  describe('when there are more than one currencies', () => {
+    beforeEach(async () => {
+      currencyService.getAll.mockReturnValue(
+        of([{ code: 'EUR' }, { code: 'USD' }, { code: 'GBP' }] as Currency[])
+      );
+      element = await fixture(
+        html`<oryx-site-currency-selector></oryx-site-currency-selector>`
+      );
+    });
+
+    it('should show the currency selector', () => {
+      expect(element).toContainElement('button');
+    });
+
+    describe('and a currency is selected', () => {
+      beforeEach(async () => {
+        element = await fixture(
+          html`<oryx-site-currency-selector></oryx-site-currency-selector>`
+        );
+        const usdOption = element.shadowRoot?.querySelector(
+          'oryx-option[value=USD]'
+        );
+        usdOption?.dispatchEvent(new Event('click'));
+      });
+
+      it('should change set the currency', () => {
+        expect(currencyService.set).toHaveBeenCalledWith('USD');
+      });
+    });
+
+    describe('and the language is Spanish', () => {
+      beforeEach(async () => {
+        localeService.get.mockReturnValue(of('es'));
+        element = await fixture(
+          html`<oryx-site-currency-selector></oryx-site-currency-selector>`
+        );
+      });
+
+      it('should render the currency selector', () => {
+        const button = element.shadowRoot?.querySelector('button');
+        expect(button).toBeDefined();
+        expect(button?.textContent).toContain('EUR');
+      });
+
+      it('should render the EUR option', () => {
+        const eur = element.shadowRoot?.querySelector('oryx-option[value=EUR]');
+        expect(eur).toBeDefined();
+        expect(eur?.textContent).toContain('euro');
+      });
+
+      it('should render the USD option', () => {
+        const eur = element.shadowRoot?.querySelector('oryx-option[value=USD]');
+        expect(eur).toBeDefined();
+        expect(eur?.textContent).toContain('dólar estadounidense');
+      });
+
+      it('should render the GBP option', () => {
+        const eur = element.shadowRoot?.querySelector('oryx-option[value=GBP]');
+        expect(eur).toBeDefined();
+        expect(eur?.textContent).toContain('libra esterlina');
+      });
+    });
+
+    describe('and the language is German', () => {
```

Comment:
what is the benefit of testing two languages? Looks like it is enough to test one language

Reply:
Just to ensure we do react to the current locale, rather than falling back to a language. But, that being said, agreed, we can only test Spanish, as this is not likely to be our default. Will change it.